### PR TITLE
Update enum values to match cusparse

### PR DESCRIPTION
**Pull request by Angeloo01**
resolves SWDEV-377947

Proposed changes:
 - update hipsparse enum numeric values to match cusparse numeric values

### CODE REVIEW

**Comment by jsandham** [comment]
Note: I think this is necessary so that if a user has cusparse code like:

int index = static_cast<int>(CUSPARSE_SPMV_ALG1);

and they convert their code to HIP, they get the same integer values.

File: library/include/hipsparse.h [hunk 1]
```diff
@@ -8930,11 +8930,11 @@ typedef enum
...
-    HIPSPARSE_SPMV_ALG_DEFAULT = 4,
-    HIPSPARSE_SPMV_COO_ALG1    = 5,
-    HIPSPARSE_SPMV_COO_ALG2    = 6,
-    HIPSPARSE_SPMV_CSR_ALG1    = 7,
-    HIPSPARSE_SPMV_CSR_ALG2    = 8
+    HIPSPARSE_SPMV_ALG_DEFAULT = 0,
+    HIPSPARSE_SPMV_COO_ALG1    = 1,
+    HIPSPARSE_SPMV_CSR_ALG1    = 2,
+    HIPSPARSE_SPMV_CSR_ALG2    = 3,
+    HIPSPARSE_SPMV_COO_ALG2    = 4
```
Comment (ntrost57) [inline library/include/hipsparse.h:8937]: I do not think this is valid?

File: library/include/hipsparse.h [hunk 2]
```diff
@@ -8962,29 +8962,29 @@ typedef enum
...
-    HIPSPARSE_SPMM_CSR_ALG3         = 13
+    HIPSPARSE_SPMM_ALG_DEFAULT      = 0,
+    HIPSPARSE_SPMM_COO_ALG1         = 1,
+    HIPSPARSE_SPMM_COO_ALG2         = 2,
+    HIPSPARSE_SPMM_COO_ALG3         = 3,
+    HIPSPARSE_SPMM_COO_ALG4         = 5,
+    HIPSPARSE_SPMM_CSR_ALG1         = 4,
+    HIPSPARSE_SPMM_CSR_ALG2         = 6,
+    HIPSPARSE_SPMM_CSR_ALG3         = 12,
+    HIPSPARSE_SPMM_BLOCKED_ELL_ALG1 = 13
```
Comment (ntrost57) [inline library/include/hipsparse.h:8973]: same here
I think cusparse removed parts of it in their latest release and renumbered.

File: library/include/hipsparse.h (hunk 1 above)
Comment (jsandham) [reply library/include/hipsparse.h:8937]: At first I thought it was  typo in cusparse, but they do define the enum using duplicate integers. This apparently is valid and I guess is used to indcate when two enums map to the same algorithm.

Comment (Angeloo01) [reply library/include/hipsparse.h:8937]: I believe this just indicates that the enum elements with the same value are equal and can be used interchangeably. For example, when used in a switch statement, only one of the elements with the same value can have a case statement. The other element that doesn't have an explicit case is mapped onto the element with the same value.

**Review by jsandham — APPROVED** [review]

**Review by ntrost57 — APPROVED** [review]

**Comment by ntrost57** [comment]
@YvanMokwinski review required

**Review by YvanMokwinski — APPROVED** [review]

**Pull request merged**